samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java

```diff
@@ -575,6 +576,15 @@ public void onStopContainerError(ContainerId containerId, Throwable t) {
     }
   }
 
+  @Override
+  public boolean isResourceExpired(SamzaResource resource) {
+    /**
+     * Time from which resource was allocated + 1 min (to account for clock skew) > Yarn Expiry Timeout
+     */
+    return System.currentTimeMillis() - resource.getTimestamp() + Duration.ofMinutes(1).toMillis()
+        > YarnConfiguration.DEFAULT_RM_CONTAINER_ALLOC_EXPIRY_INTERVAL_MS;
```

Comment:
Could DEFAULT_RM_CONTAINER_ALLOC_EXPIRY_INTERVAL_MS be a Duration?

Duration.between(resource.getTS, Instant.now()).compareTo(DEFAULT_RM_CONTAINER_ALLOC_EXPIRY_INTERVAL_MS) > 0;

Reply:
will make this cleaner